build/configs/esp32_DevKitC/scripts/esp32.template

```diff
@@ -52,10 +52,13 @@ MEMORY
   /* Shared data RAM, excluding memory reserved for ROM bss/data/stack.
    * Enabling Bluetooth & Trace Memory features in menuconfig will decrease
    * the amount of RAM available.
+   * Note: Length of this section *should* be 0x50000, and this extra DRAM is available
+   * in heap at runtime. However due to static ROM memory usage at this 176KB mark, the
+   * additional static memory temporarily cannot be used.
    */
 
   dram0_0_seg (RW) :     org = 0x3ffb0000 + CONFIG_ESP32_BT_RESERVE_DRAM,
-                         len = 0x50000 - CONFIG_ESP32_TRACEMEM_RESERVE_DRAM - CONFIG_ESP32_BT_RESERVE_DRAM
+                         len = 0x30000 - CONFIG_ESP32_TRACEMEM_RESERVE_DRAM - CONFIG_ESP32_BT_RESERVE_DRAM
```

Comment:
1. Does `due to static ROM memory usage at this 176KB mark` mean `CONFIG_ESP32_TRACEMEM_RESERVE_DRAM + CONFIG_ESP32_BT_RESERVE_DRAM`?
2.  If 1st is right, why this is changed? As you wrote in new comments, 0x50000 looks right. 
```
Length of this section *should* be 0x50000,
```

Reply:
CONFIG_ESP32_TRACEMEM_RESERVE_DRAM and CONFIG_ESP32_BT_RESERVE_DRAM
is not mean static ROM memory. it mean some reseved(specific) memory for debug and Bluetooth.

the meaning is that part of extral DRAM memory will be accessed by ROM code (mainly 1st bootloader code) 
if not limited it(that is give hardware DRAM size 0x50000 ), during booting,  image .data or .bss will overlap this exral DRAM, then rom code will access our image's data, then lead to unexpected result, like crasing during booting.
 
This modification is ported to ESP-IDF link script.